docs/DeveloperGuide.md

```diff
@@ -4,9 +4,118 @@
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Classes: overview
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### `Activity` and child classes
+The `Activity` class serves as a parent class to `Exercise`, `ExerciseLog`, `Workout`, `WorkoutLog` and `Day` classes for the ease of usage of `ActivityManager` classes (see below).
+
+![ActivityClassDiagram](diagrams/ActivityClassDiagram.png)
```

Comment:
Is the Parser class missing an activation bar?
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/110594538/4abfd1ad-18d7-449e-82f0-d72c62bdb73d)